auslib/web/admin/swagger/api.yaml

```diff
@@ -0,0 +1,441 @@
+            properties:
+              count:
+                type: integer
+                format: int32
+                example: 1
+              rules:
+                type: array
+                items:
+                  $ref: '#/definitions/RulesBaseGET'
+                example:
+                  - alias: "esr45-localtest"
+                    backgroundRate: 100
+                    buildID: null
+                    buildTarget: null
+                    channel: "esr-localtest"
+                    comment: "esr45 esr-localtest rule"
+                    data_version: 22
+                    distVersion: null
+                    distribution: null
+                    fallbackMapping: null
+                    headerArchitecture: null
+                    locale: null
+                    mapping: "Firefox-45.7.0esr-build1"
+                    osVersion: null
+                    priority: 43
+                    product: "Firefox"
+                    rule_id: 308
+                    systemCapabilities: null
+                    update_type: "minor"
+                    version: null
+                    whitelist: null
+          #headers : [Optional]
+
+    post:
+      tags:
+        - Rules
+      summary: creates a Rule
+      description: creates a new Rule in Balrog's DB
+      consumes:
+        - application/json
+      produces:
+        - text/html
+      deprecated: false
+      operationId: auslib.web.admin.views.mapper.rules_post
+      parameters:
+        - name: rule
+          in: body
+          description: New Rule object data
+          required: true
+          schema:
+            allOf:
+              - $ref: '#/definitions/CSRFModel'
+              - $ref: '#/definitions/RulesBase'
+
+            # will fail testMissingField test in test_rules.py if test is not modified.
+            # Because only outputs first missing field : backgroundRate
+            # ignores further check i.e. check for priority and returns error.
+            required:
+              - backgroundRate
+              - priority
+
+      responses:
+        '200':
+          description: Created a new rule
+          schema:
+            type: string
+            example: "514"
+        '400':
+          description: Requires remote username and password in basic authentication
+          schema:
+            type: string
+            example: "{'error': 'invalid form values'}"
+          examples:
+            application/json: "{'error': 'Invalid Form Values'}"
+        default:
+          description: Requires remote username and password in basic authentication
+          examples:
+            text/html: "Error: REMOTE_USERNAME required"
+
+
+definitions:
+  CSRFModel:
+    title: CSRF Token
+    description: CSRF Token in POST
+    type: object
+
+    #Existing Test cases fail when we don't provide this value. Hence keep it out of required section.
+    #Make it optional.
+    #required:
+    #- csrf_token
```

Comment:
It's important that csrf_token is required in production, so we'll need some sort of workaround for this.